src/linear_algebra/tensor_product/matrix.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+
+import data.matrix.kronecker
+import linear_algebra.matrix.to_lin
+import linear_algebra.tensor_product.basis
+
+/-!
+# Connections between `tensor_product` and `matrix`
+
+This file contains results about the matrices corresponding to maps between tensor product types,
+where the correspondance is induced by `basis.tensor_product`
+
+Notably, `tensor_product.to_matrix_map` shows that taking  the tensor product of linear maps is
+equivalent to taking the kronecker product of their matrix representations.
+-/
+
+variables {R : Type*} {M N P M' N' : Type*} {ι κ τ ι' κ' : Type*}
+variables [decidable_eq ι] [decidable_eq κ] [decidable_eq τ]
+variables [fintype ι] [fintype κ] [fintype τ] [fintype ι'] [fintype κ']
+variables [comm_ring R]
+variables [add_comm_group M] [add_comm_group N] [add_comm_group P]
+variables [add_comm_group M'] [add_comm_group N']
+variables [module R M] [module R N] [module R P] [module R M'] [module R N']
+variables (bM : basis ι R M) (bN : basis κ R N) (bP : basis τ R P)
+variables (bM' : basis ι' R M') (bN' : basis κ' R N')
+
+open_locale kronecker
+open matrix linear_map
+
+/-- The linear map built from `tensor_product.map` corresponds to the matrix built from
+`matrix.kronecker`. -/
+lemma tensor_product.to_matrix_map (f : M →ₗ[R] M') (g : N →ₗ[R] N') :
+  to_matrix (bM.tensor_product bN) (bM'.tensor_product bN') (tensor_product.map f g)
+    = to_matrix bM bM' f ⊗ₖ to_matrix bN bN' g :=
```

Comment:
Is it worth turning this around and simping it?

Reply:
Turning it around feels weird to me, usually our lemmas about a function push it to the inside.

I considered adding `simp` but wasn't convinced; do you feel strongly one way or the other about it?

Reply:
I was considering turning it around because the RHS seems syntactically more general than the LHS, so I suspect it would get rewritten more often reversed. I don't feel strongly about the simp, but thought you might want to consider because it looks eligible.

Reply:
I'd claim the RHS is simpler because it refers to two functions not three.

Reply:
I'll express a weak preference for keeping it as-is for symmetry with `to_matrix_comm`, which is more obviously in the correct direction IMO since `1.submatrix` is easy to evaluate.